src/azure-cli/azure/cli/command_modules/vm/_help.py

```diff
@@ -2138,8 +2138,18 @@
 
 helps['vm secret add'] = """
 type: command
-short-summary: Add a secret to a VM.
+short-summary: Add a secret to a VM. To install certificates on a virtual machine it is recommended to use the [Azure Key Vault virtual machine extension for Linux](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-linux) or the [Azure Key Vault virtual machine extension for Windows](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-windows) instead of `az vm secret add`. 
 examples:
+  - name: Add a secret to a VM using the Azure Key Vault virtual machine extension (recommended).
+    text: |
+        # Start the deployment
+        az vm extension set -n "KeyVaultForLinux" `
+        --publisher Microsoft.Azure.KeyVault `
+        -g "<resourcegroup>" `
+        --vm-name "<vmName>" `
+        --version 2.0 `
+        --settings '{\"secretsManagementSettings\": { \"pollingIntervalInS\": \"<pollingInterval>\", \"certificateStoreName\": \"<certStoreName>\", \"certificateStoreLocation\": \"<certStoreLoc>\", \"observedCertificates\": [\" <observedCert1> \", \" <observedCert2> \"] }}'
+    crafted: true
```

Comment:
If the help is handwritten and not provided by Aladdin, please remove the `crafted: true` tag